SIT.Manager.Avalonia/ViewModels/Installation/PatchViewModel.cs

```diff
@@ -0,0 +1,172 @@
+        _downloadProgress.ProgressChanged += DownloadProgress_ProgressChanged;
+        _extractionProgress.ProgressChanged += ExtractionProgress_ProgressChanged;
+    }
+
+    private void CopyProgress_ProgressChanged(object? sender, double e)
+    {
+        CopyProgressPercentage = e;
+    }
+
+    private void DownloadProgress_ProgressChanged(object? sender, double e)
+    {
+        DownloadProgressPercentage = e * 100;
+    }
+
+    private void ExtractionProgress_ProgressChanged(object? sender, double e)
+    {
+        ExtractionProgressPercentage = e;
+    }
+
+    private async Task RunPatcher()
+    {
+        string[] files = Directory.GetFiles(CurrentInstallProcessState.EftInstallPath, "Patcher.exe", new EnumerationOptions() { MatchCasing = MatchCasing.CaseInsensitive, MaxRecursionDepth = 0 });
+        if (files.Length == 0)
+        {
+            // TODO handle this: return $"Patcher.exe not found in {CurrentInstallProcessState.EftInstallPath}";
+        }
+        string patcherPath = files[0];
+
+        EmbeddedProcessWindow embeddedProcessWindow = new(_installerService.CreatePatcherProcess(patcherPath));
+        await embeddedProcessWindow.StartProcess();
+
+        // TODO get window embedding working on linux - maybe we just add the patcher as a git module? and then embed the window of the patcher itself?
+        // Window embedding only seems to work properly on Windows 
+        // More info here: https://github.com/AvaloniaUI/Avalonia/issues/10354 and https://github.com/AvaloniaUI/Avalonia/issues/6124
+        // So we will only do this on Windows for now until we spend some time to get it working nicely on Linux.
+        if (OperatingSystem.IsWindows())
+        {
+            EmbeddedPatcherWindow = embeddedProcessWindow;
+            await EmbeddedPatcherWindow.WaitForExit();
+        }
+        else
+        {
+            await embeddedProcessWindow.StartProcess();
+        }
+
+        _patcherResultMessages.TryGetValue(EmbeddedPatcherWindow.ExitCode, out string? patcherResult);
+        _logger.LogInformation($"RunPatcher: {patcherResult}");
+
+        int exitCode = EmbeddedPatcherWindow.ExitCode;
+        EmbeddedPatcherWindow = null;
+
+        // Success exit code
+        if (exitCode == 10)
+        {
+            if (File.Exists(patcherPath))
+            {
+                File.Delete(patcherPath);
+            }
+
+            string patcherLog = Path.Combine(CurrentInstallProcessState.EftInstallPath, "Patcher.log");
+            if (File.Exists(patcherLog))
+            {
+                File.Delete(patcherLog);
+            }
+
+            string akiPatchesDir = Path.Combine(CurrentInstallProcessState.EftInstallPath, "Aki_Patches");
+            if (Directory.Exists(akiPatchesDir))
+            {
+                Directory.Delete(akiPatchesDir, true);
+            }
+
+            ProgressInstall();
+        }
+        else
+        {
+            // TODO report error here somehow
```

Comment:
Can we chuck a way to reset the install steps so this doesn't softlock the manager?